src/TourPortal.js

```diff
@@ -140,6 +140,12 @@ class TourPortal extends Component {
     if (isOpen) {
       this.close()
     }
+    if (this.state.observer) {
+      this.state.observer.disconnect()
+      this.setState({
```

Comment:
You could avoid calling `setState` in cWU, more info [here](https://reactjs.org/docs/react-component.html#componentwillunmount)